src/cargo/ops/cargo_install.rs

```diff
@@ -422,7 +423,7 @@ impl<'gctx> InstallablePackage<'gctx> {
         for &(bin, src) in binaries.iter() {
             let dst = staging_dir.path().join(bin);
             // Try to move if `target_dir` is transient.
-            if !self.source_id.is_path() && fs::rename(src, &dst).is_ok() {
+            if (!self.source_id.is_path() && fs::rename(src, &dst).is_ok()) || dry_run {
```

Comment:
Shouldn't this be `if dry_run ||` so we don't do the `fs::rename`?

Reply:
Absolutely, I fixed this